config-processor.js

```diff
@@ -0,0 +1,156 @@
+ *
+ * @param {PluginManager} pluginManager The plugin manager
+ * @param {PluginList} orgPlugins The list of plugins configured on the org
+ * @param {PluginList} repoIncludeList A list of plugins to merge into the config
+ * @param {OnInvalidPluginCallback} [onInvalidPlugin] Function to call when an invalid plugin is encountered
+ * @returns {PluginList} The filtered list of plugins
+ */
+// eslint-disable-next-line max-statements, complexity
+function applyIncludeList(pluginManager, orgPlugins, repoIncludeList, onInvalidPlugin) {
+  const plugins = deepClone(orgPlugins);
+  for (const pluginToInclude of repoIncludeList) {
+    const pluginName = typeof pluginToInclude === 'string' ? pluginToInclude : pluginToInclude.plugin;
+    const pluginInstance = pluginManager[pluginName];
+    if (!pluginInstance) {
+      onInvalidPlugin && onInvalidPlugin(pluginName);
+      // eslint-disable-next-line no-continue
+      continue;
+    }
+
+    if (typeof pluginToInclude === 'string' &&
+      !plugins.find(p => p === pluginToInclude ||
+        (/** @type {Plugin} */(p).plugin && /** @type {Plugin} */(p).plugin === pluginToInclude))) {
+      // If the entry in the include list is just a string, we can simply add it if it's not present.
+      plugins.push(pluginToInclude);
+    } else if (typeof pluginToInclude === 'object' && pluginToInclude.plugin) {
+      // If the entry in the include list is an object, we have to be more careful.
+      const existingPlugin = plugins.find(
+        p => p === pluginToInclude.plugin ||
+        (/** @type {Plugin} */(p).plugin && /** @type {Plugin} */(p).plugin === pluginToInclude.plugin));
+      if (!existingPlugin) {
+        // If it's not in the existing list, just add it directly
+        plugins.push(pluginToInclude);
+      } else if (typeof existingPlugin === 'string' || !existingPlugin.config) {
+        // If it is in the existing list but only as a string, we can just replace the string with the object.
+        // Same story if the existing plugin has no config field set.
+        const idx = plugins.indexOf(existingPlugin);
+        plugins.splice(idx, 1, pluginToInclude);
+      } else {
+        // If it is in the existing list as an object, and the plugin to include has config specified, we have to merge
+        // the config
+        existingPlugin.config = pluginInstance.mergeConfig(existingPlugin.config, pluginToInclude.config);
+      }
+    }
+  }
+
+  return plugins;
+}
+
+/**
+ * Apply the exclude list of plugin names and return the filtered plugin list
+ *
+ * @param {PluginList} orgPlugins The list of plugins configured on the org
+ * @param {string[]} repoExcludeList A list of plugin names to exclude
+ * @returns {PluginList} The filtered list of plugins
+ */
+function applyExcludeList(orgPlugins, repoExcludeList) {
+  return orgPlugins.filter(plugin => {
+    if (typeof plugin === 'string') {
+      return !~repoExcludeList.indexOf(plugin);
```

Comment:
I see you are using object rest spread above which leads me to believe that this does not need to be backwards compatible with older JavaScript VMs. 

As such I would suggest using `.includes` instead of the `indexOf` bitshifting shorthand for readability.

Reply:
Ah I was looking at this earlier and was going to suggest `.includes` but got distracted!